src/cmd/groff/contrib/groffer/roff2.man
.ig
roff2@ROFF2MODE@.1 - man page for roff2@ROFF2MODE@ (section 1).

Source file position: <groff_source_top>/contrib/groffer/roff2.man
Installed position:   $prefix/share/man/man1/roff2@ROFF2MODE@.1

Last update: 2 Feb 2007
Simplified to use .SY/.OP/.YS
..
.
.
.de au
This file was written by Bernd Warken.
..
.
.
.de co
Copyright (C) 2006, 2007 Free Software Foundation, Inc.
.
.P
This file is part of
.IR \%groffer ,
which is part of
.IR \%groff ,
a free software project.
.
You can redistribute it and/or modify it under the terms of the
.nh
.B "GNU General Public License"
.hy
as published by the
.nh
.BR "Free Software Foundation" ,
.hy
either version\~2, or (at your option) any later version.
.
.P
You should have received a copy of the \f[CR]GNU General Public
License\f[] along with
.IR groff ,
see the files \%\f[CB]COPYING\f[] and \%\f[CB]LICENSE\f[] in the top
directory of the
.I groff
source package.
.
Or read the
.I man\~page
.BR gpl (1).
You can also write to the
.nh
.B "Free Software Foundation, 51 Franklin St - Fifth Floor, Boston,"
.BR "MA 02110-1301, USA" .
.hy
..
.
.
.TH roff2@ROFF2MODE@ @MAN1EXT@ "@MDATE@" "Groff Version @VERSION@"
.SH NAME
roff2@ROFF2MODE@ \- transform roff code into @ROFF2MODE@ mode
.
.
.\" --------------------------------------------------------------------
.SH "SYNOPSIS"
.
.SY roff2@ROFF2MODE@
.RI [ \%groffer_option .\|.\|.\&]
.OP \-\-
.RI [ \%filespec .\|.\|.\&]
.
.SY roff2@ROFF2MODE@
.B \-h
|
.B \-\-help
.
.SY roff2@ROFF2MODE@
.B \-v
|
.B \-\-version
.YS
.
.P
The options
.B \-v
and
.B \%\-\-version
print the version information of the program to standard output and exit.
.
The options
.B \-h
and
.B \-\-help
print a usage information of the program to standard output and stop
the program instantly.
.
.
.P
All other options are assumed to be
.B \%groffer
options.
.
They are internally passed to
.BR \%groffer .
They override the behavior of the program.
.
The options are optional, they can be omitted.
.
.
.P
The
.I \%filespec
arguments correspond to the
.I \%filespec
arguments of
.BR \%groffer .
So they are either the names of existing, readable files or
.B \-
for standard input, or the name of a man page or a
.BR \%groffer (@MAN1EXT@)
man page search pattern.
.
If no
.I \%filespec
is specified standard input is assumed automatically.
.
.
.\" --------------------------------------------------------------------
.SH DESCRIPTION
.
.B \%roff2@ROFF2MODE@
transforms
.I roff
code into
.ie '@ROFF2MODE@'x' \{\
.  I X
mode corresponding to the
.  I groff
devices
.  BR X *;
this mode is suitable for
.  BR \%gxditview (@MAN1EXT@).
.\}
.el \{\
.  I \%@ROFF2MODE@
mode.
.\}
.
Print the result to standard output.
.
.
.P
There are more of these programs for generating other formats of
.I \%roff
input.
.
.if !'@ROFF2MODE@'dvi' \{\
.  TP
.  BR \%roff2dvi (@MAN1EXT@)
is for
.  I dvi
mode.
.\}
.
.if !'@ROFF2MODE@'html' \{\
.  TP
.  BR \%roff2html (@MAN1EXT@)
generates
.  I html
output.
.\}
.
.if !'@ROFF2MODE@'pdf' \{\
.  TP
.  BR \%roff2pdf (@MAN1EXT@)
outputs
.  I pdf
mode.
.\}
.
.if !'@ROFF2MODE@'ps' \{\
.  TP
.  BR \%roff2ps (@MAN1EXT@)
prints
.  I \%PostScript
format to standard output.
.\}
.
.if !'@ROFF2MODE@'text' \{\
.  TP
.  BR \%roff2text (@MAN1EXT@)
generates text output in the
.  B groff
device
.  IR latin1 .
.\}
.
.if !'@ROFF2MODE@'x' \{\
.  TP
.  BR \%roff2x (@MAN1EXT@)
prints the output in the
.  B groff
device
.  B X
that is suitable for programs like
.  BR \%gxditview (@MAN1EXT@)
or
.  BR \%xditview (1).
.\}
.
.
.\" --------------------------------------------------------------------
.SH "SEE ALSO"
.\" --------------------------------------------------------------------
.
.BR \%groff (@MAN1EXT@),
.BR \%groffer (@MAN1EXT@),
.if !'@ROFF2MODE@'dvi' \
.  BR \%roff2dvi (@MAN1EXT@),
.if !'@ROFF2MODE@'html' \
.  BR \%roff2html (@MAN1EXT@),
.if !'@ROFF2MODE@'pdf' \
.  BR \%roff2pdf (@MAN1EXT@),
.if !'@ROFF2MODE@'ps' \
.  BR \%roff2ps (@MAN1EXT@),
.if !'@ROFF2MODE@'text' \
.  BR \%roff2text (@MAN1EXT@),
.if !'@ROFF2MODE@'x' \
.  BR \%roff2x (@MAN1EXT@),
.BR \%gxditview (@MAN1EXT@).
.
.
.\" --------------------------------------------------------------------
.SH "AUTHOR"
.\" --------------------------------------------------------------------
.au
.
.
.\" --------------------------------------------------------------------
.SH "COPYING"
.\" --------------------------------------------------------------------
.co
.
.
.\" --------------------------------------------------------------------
.\" Emacs settings
.\" --------------------------------------------------------------------
.
.\" Local Variables:
.\" mode: nroff
.\" End:


src/cmd/groff/contrib/groffer/shell/roff2.sh
#! /bin/sh

# roff2* - transform roff files into other formats

# Source file position: <groff-source>/contrib/groffer/shell/roff2.sh
# Installed position: <prefix>/bin/roff2*

# Copyright (C) 2006 Free Software Foundation, Inc.
# Written by Bernd Warken

# Last update: 11 Nov 2006

# This file is part of `groffer', which is part of `groff'.

# `groff' is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2, or (at your option)
# any later version.

# `groff' is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with `groff'; see the files COPYING and LICENSE in the top
# directory of the `groff' source.  If not, write to the Free Software
# Foundation, 51 Franklin St - Fifth Floor, Boston, MA 02110-1301,
# USA.

########################################################################


##############
# echo1 (<text>*)
#
# Output to stdout with final line break.
#
# Arguments : arbitrary text including `-'.
#
echo1()
{
  cat <<EOF
$@
EOF
} # echo1()


##############
# echo2 (<text>*)
#
# Output to stderr with final line break.
#
# Arguments : arbitrary text including `-'.
#
echo2()
{
  cat >&2 <<EOF
$@
EOF
} # echo2()


error_no_groffer='no';
error_no_groffer() {
  if test _"$error_no_groffer" = _yes
  then
    return;
  fi;
  error_no_groffer='yes';
  echo2 "$name: groffer is not available.";
}


error_no_options='no';
error_no_options() {
  if test _"$error_no_options" = _yes
  then
    return;
  fi;
  error_no_groffer='yes';
  echo2 "$name: groffer options are not allowed.";
}


usage_with_groffer() {
  cat <<EOF
usage: $Name [option]... [--] [filespec]...

-h | --help     print usage information
-v | --version  print version information

All other options are arbitrary options of "groffer"; the options
override the behavior of this program.

"filespec"s are the same as in "groffer": either the names of
existing, readable files or "-" for standard input or a search pattern
for man pages.  No "filespec" assumes standard input automatically.
EOF
}

usage_without_groffer() {
  cat <<EOF
usage: $Name [option]... [--] [filespec]...

-h | --help     print usage information
-v | --version  print version information

No other options are allowed because "groffer" is not available.

The only "filespec"s allowed are the names of existing, readable files
or "-" for standard input.  No "filespec" assumes standard input
automatically.
EOF
}


where_is_prog() {
  for p in `echo $PATH|sed "s/:/ /g"`
  do
    f="${p}/$1";
    if test -f "$f" && test -x "$f"
    then
      echo1 "$f";
      return;
    fi;
  done;
}


########################################################################

export NULL_DEV;
if test -c /dev/null
then
  NULL_DEV='/dev/null';
else
  NULL_DEV='NUL';
fi;

name="$(echo1 "$0" | sed 's|^.*//*\([^/]*\)$|\1|')";

case "$name" in
roff2[a-z]*)
  mode="$(echo1 "$name" | sed 's/^roff2//')";
  ;;
*)
  echo2 "wrong program name: $name";
  exit 1;
  ;;
esac;

groff_version="$(groff --version 2>$NULL_DEV)";
if test $? -gt 0
then
  echo2 "$name error: groff does not work.";
  exit 1;
fi;
groffer_version="$(groffer --version 2>$NULL_DEV)";
if test $? -gt 0
then
  has_groffer='no';
else
  has_groffer='yes';
fi;

if test _"${has_groffer}" = _yes
then
  for i
  do
    case $i in
    -v|--v|--ve|--ver|--vers|--versi|--versio|--version)
      echo1 "$name in $groffer_version";
      exit 0;
      ;;
    -h|--h|--he|--hel|--help)
      usage_with_groffer;
      exit 0;
      ;;
    esac;
  done;
  groffer --to-stdout --$mode "$@";
else				# not has_groffer
  reset=no;
  double_minus=no;
  for i
  do
    if test _"${reset}" = _no
    then
      set --;
      reset=yes;
    fi;
    if test _"${double_minus}" = _yes
    then
      set -- "$@" "$i";
      continue;
    fi;
    case "$i" in
    --)
      double_minus=yes;
      continue;
      ;;
    -)
      set -- "$@" '-';
      continue;
      ;;
    -v|--v|--ve|--ver|--vers|--versi|--versio|--version)
      echo1 "$name in $groff_version";
      exit 0;
      ;;
    -h|--h|--he|--hel|--help)
      usage_without_groffer;
      exit 0;
      ;;
    -*)
      error_no_groffer;
      error_no_options;
      ;;
    *)
      if test -f "$i" && test -r "$i"
      then
        set -- "$@" "$i";
      else
        error_no_groffer;
        echo2 "$i is not an existing, readable file.";
      fi;
      continue;
      ;;
    esac;
  done;				# for i

  if test $# -eq 0
  then
    set -- '-';
  fi;
  has_stdin=no;
  for i
  do
    case "$i" in
    -)
      has_stdin=yes;
      break;
      ;;
    esac;
  done;

  if test _$has_stdin = _yes
  then
    umask 0077;
    tempdir='';
    for d in "${GROFF_TMPDIR}" "${TMPDIR}" "${TMP}" "${TEMP}" \
           "${TEMPDIR}" "${HOME}"'/tmp' '/tmp' "${HOME}" '.'
    do
      if test _"$d" = _ || ! test -d "$d" || ! test -w "$d"
      then
        continue;
      fi;
      case "$d" in
      */)
        tempdir="$d";
        ;;
      *)
        tempdir="$d"'/';
        ;;
      esac;
    done;
    if test _$tempdir = _
    then
      echo2 "${name}: could not find a temporary directory."
      exit 1;
    fi;
    stdin=${tempdir}${name}_$$;
    if test -e "$stdin"
    then
      rm -f "$stdin";
      n=0;
      f="${stdin}_$n";
      while test -e "$f"
      do
	rm -f "$f";
        if ! test -e "$f"
        then
          break;
        fi;
        n="$(expr $n + 1)";
        f="${stdin}_$n";
      done;
      stdin="$f";
    fi;
    reset=no;
    for i
    do
      if test _"${reset}" = _no
      then
        set --;
        reset=yes;
      fi;
      case "$i" in
      -)
        set -- "$@" "$stdin";
        ;;
      *)
        set -- "$@" "$i";
        ;;
      esac;
    done;
    cat>"$stdin";
  fi;				# if has_stdin

  case "$mode" in
  x)
    groff_options='-TX75-12 -Z';
    ;;
  text)
    groff_options='-Tlatin1';
    ;;
  pdf)
    ps2pdf="$(where_is_prog ps2pdf)";
    if test _"$ps2pdf" = _
    then
      ps2pdf="$(where_is_prog gs)";
      if test _"$ps2pdf" = _
      then
        echo2 "$name: cannot transform to pdf format.";
        exit 1;
      fi;
      ps2pdf="$ps2pdf -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite '-sOutputFile=- -c save pop -f -";
    else
      ps2pdf="$ps2pdf -";
    fi;
    grog="$(grog -Tps "$@")";
    eval $grog | ${ps2pdf};
    exit $?;
    ;;
  *)
    groff_options="-T$mode";
    ;;
  esac;
  grog="$(grog $groff_options "$@")";
  eval $grog;
  exit $?;
fi;				# not has_groffer
